src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -705,9 +705,27 @@ CHIP_ERROR AccessControlAttribute::WriteExtension(const ConcreteDataAttributePat
 
 AccessControlAttribute gAttribute;
 
+class : public FabricTableDelegate
+{
+public:
+    void OnFabricDeletedFromStorage(CompressedFabricId compressedId, FabricIndex fabricIndex) override
+    {
+        auto & storage = Server::GetInstance().GetPersistentStorage();
+        DefaultStorageKeyAllocator key;
+        storage.SyncDeleteKeyValue(key.AccessControlExtensionEntry(fabricIndex));
+    }
+    void OnFabricRetrievedFromStorage(FabricInfo * fabricInfo) override {}
+    void OnFabricPersistedToStorage(FabricInfo * fabricInfo) override {}
+
+} fabricTableDelegate;
+
 } // namespace
 
 void MatterAccessControlPluginServerInitCallback()
 {
+    ChipLogProgress(DataManagement, "AccessControlCluster: initializing");
+
     registerAttributeAccessOverride(&gAttribute);
+
+    Server::GetInstance().GetFabricTable().AddFabricDelegate(&fabricTableDelegate);
```

Comment:
How many slots are there of these? Could we run out and not notice due to not being able to return errors from this init?

Reply:
Linked list.